### PR TITLE
#5382 Use custom checkbox and radiobutton in listview config

### PR DESCRIPTION
**Pull request by poornimanayar**
### Prerequisites

- [ ] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes <https://github.com/umbraco/Umbraco-CMS/issues/5382>

Go to a listview config, the order direction radio button must conform to the custom radio butons. The checkboxes for layout must conform to the custom checkboxes. The functionality should be same as that of now.

image

image

Poornima

### CODE REVIEW

**Comment by emmaburstow** [comment]
Hey again @poornimanayar 

Thanks for the work. We'll let you know if we need anything!

Em

File: src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/layouts.prevalues.html [hunk 1]
```diff
@@ -29,7 +29,7 @@
          </div>
 
          <div>
-            <input ng-if="layout.isSystem === 1" type="checkbox" ng-model="layout.selected" />
+             <umb-checkbox model="layout.selected" class="mce-cmd" />
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/layouts.prevalues.html]: @poornimanayar I'm missing the `ng-if="layout.isSystem === 1"` check here. We should preserve this.

File: src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/orderDirection.prevalues.html [hunk 2]
```diff
@@ -1,15 +1,8 @@
...
-            <input type="radio" name="orderDirection" value="asc" ng-model="model.value" />
-            Ascending <code>[a-z]</code>
-        </label>
-        <label class="radio">
-            <input type="radio" name="orderDirection" value="desc" ng-model="model.value" />
-            Descending <code>[z-a]</code>
-        </label>
-
-        <span ng-messages="listViewOrderDirectionForm.orderDirection.$error" show-validation-on-submit >
+        <umb-radiobutton name="orderDirection" value="asc" model="model.value" text="Ascending"></umb-radiobutton>
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/orderDirection.prevalues.html]: This removes `<code>[a-z]</code>`, which should be preserved. Same for the `Descending` radiobutton below it.

File: src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/layouts.prevalues.html (hunk 1 above)
Comment (bjarnef) [reply src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/layouts.prevalues.html]: @poornimanayar you need to update this to the following: `<umb-checkbox ng-if="layout.isSystem === 1" model="layout.selected" />`

Comment (poornimanayar) [reply src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/layouts.prevalues.html]: Thanks @bjarnef for reminding me about this PR. I had totally forgotten about this. I will look at this once I get back from the PL fest. Added to my to-do list :-)

File: src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/orderDirection.prevalues.html (hunk 2 above)
Comment (bjarnef) [reply src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/orderDirection.prevalues.html]: @poornimanayar change this part to something like this:

```
<div class="flex items-center">
    <div class="flex-inline items-center">
        <umb-radiobutton name="orderDirection" value="asc" model="model.value" text="Ascending" /><code class="ml1">[a-z]</code>
    </div>
    <div class="flex-inline items-center ml3">
        <umb-radiobutton name="orderDirection" value="desc" model="model.value" text="Descending" /><code class="ml1">[z-a]</code>
    </div>
</div>
```

File: src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/layouts.prevalues.html (hunk 1 above)
Comment (bjarnef) [reply src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/layouts.prevalues.html]: That would be great :)
I also noticed the listview layout delete/trash icon in custom listview layouts can't be focused using tab key.

Changing the following to a button element seems to fix this.

```
<i class="icon-trash" ng-click="vm.showPrompt(layout)"></i>
```

```
<button type="button" class="btn-link" ng-click="vm.showPrompt(layout)">
    <i class="icon-trash" aria-hidden="true"></i>
</button>
```

I guess this is a great task in another PR for Mr. @BatJan 😁👍🦇

Comment (BatJan) [reply src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/layouts.prevalues.html]: Uh, love the bat emoji! 😄 - Once this PR is done I will have a look at doing some fallback texts if that's what you have in mind @bjarnef ? 😃

Comment (bjarnef) [reply src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/layouts.prevalues.html]: @BatJan yes, that emoji is very cool! 🦄 
It was not much related to this PR, but just the listview layout delete buttons - only shown for custom layouts.

So this line:

```
<i class="icon-trash" ng-click="vm.showPrompt(layout)"></i>
```

needs to be changed to something like this (an anchor might also work, but I guess a `button` is ideal:

```
<button type="button" class="btn-link" ng-click="vm.showPrompt(layout)">
    <i class="icon-trash" aria-hidden="true"></i>
</button>
```

Comment (BatJan) [reply src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/layouts.prevalues.html]: @bjarnef Aaah, got you! 😉

**Comment by poornimanayar** [comment]
Hi @bjarnef ,

Many thanks for reminding me about this PR and giving me some help as well around it. I have included that in and pushed the code in :-)

Poornima

**Comment by bjarnef** [comment]
@poornimanayar I don't think you need the class `mce-cmd` on the checkbox. Further you can remove a space before `model` attrubute.

**Comment by poornimanayar** [comment]
Closing as this was already done in anotehr PR

**Pull request closed without merging**